vllm/core/scheduler.py

```diff
@@ -866,9 +876,18 @@ def _schedule_prefills(
                 curr_loras.add(lora_int_id)
             waiting_queue.popleft()
             self._allocate_and_set_running(seq_group)
-            seq_group.init_multi_step(
-                num_scheduler_steps=self._get_num_lookahead_slots(
-                    is_prefill=True) + 1)
+
+            if enable_chunking and self.scheduler_config.is_multi_step:
+                # TODO (varun) : Is this correct? We have an assert an assert
+                # here since we have already checked slots availability in
+                # `block_manager.can_allocate`
+                assert self._can_append_slots(seq_group)
```

Comment:
If we are sure about this then we could avoid the overhead of _can_append_slots

Reply:
Removed. I am fairly confident that `can_allocate` is sufficient.